### ISSUE TITLE
Clarify docs for CronWorkflow `startingDeadlineSeconds`

### ISSUE THREAD

**Issue by nhavens** [argoproj/argo-workflows#12971]
### Pre-requisites

- [X] I have double-checked my configuration
- [X] I have tested with the `:latest` image tag (i.e. `quay.io/argoproj/workflow-controller:latest`) and can confirm the issue still exists on `:latest`. If not, I have explained why, **in detail**, in my description below.
- [X] I have searched existing issues and could not find a match for this bug
- [ ] I'd like to contribute the fix myself (see [contributing guide](https://github.com/argoproj/argo-workflows/blob/main/docs/CONTRIBUTING.md))

### What happened/what did you expect to happen?

The [CronWorkflow docs](https://argo-workflows.readthedocs.io/en/latest/cron-workflows/#crash-recovery) describe `startingDeadlineSeconds` as:
> Mainly startingDeadlineSeconds can be set to specify the maximum number of seconds past the last successful run of a CronWorkflow during which a missed run will still be executed.
>
> For example, if a CronWorkflow that runs every minute is last run at 12:05:00, and the controller crashes between 12:05:55 and 12:06:05, then the expected execution time of 12:06:00 would be missed. However, if startingDeadlineSeconds is set to a value greater than 65 (the amount of time passing between the last scheduled run time of 12:05:00 and the current controller restart time of 12:06:05), then a single instance of the CronWorkflow will be executed exactly at 12:06:05.

This does not match the logic in the [implementation](https://github.com/argoproj/argo-workflows/blob/v3.5.5/workflow/cron/operator.go#L271-L278). Based on the code, the docs should be reworded to something like the following:
> Mainly startingDeadlineSeconds can be set to specify the maximum number of seconds past the last scheduled run of a CronWorkflow during which a missed run will still be executed.
>
> For example, if a CronWorkflow that runs every minute is last run at 12:05:00, and the controller crashes between 12:05:55 and 12:06:05, then the expected execution time of 12:06:00 would be missed. However, if startingDeadlineSeconds is set to a value greater than 5 (the amount of time passing between the last scheduled run time of 12:06:00 and the current time of 12:06:05), then a single instance of the CronWorkflow will be executed exactly at 12:06:05.

Similar changes should be made elsewhere, as the comparison is not against the last successful run, but rather the last desired/expected/scheduled run time.

tl;dr `startingDeadlineSeconds` should be clearly documented as the "grace period" beyond the scheduled run during which a CronWorkflow may still be executed.

### Version

v3.5.5

### Paste a small workflow that reproduces the issue. We must be able to run the workflow; don't enter a workflows that uses private images.

```YAML
n/a
```

### Logs from the workflow controller

```text
n/a
```

### Logs from in your workflow's wait container

```text
n/a
```

**Comment by Joibel** [comment]
PRs welcome on this.

**Comment by parambath92** [comment]
Hello @Joibel, I am interested in picking this up if nobody else is working on it.

**Comment by Nidicar29** [comment]
https://www.youtube.com/@Nidicar29oficial

Soy nidicar29 subscripción Youtube de channel 
![Screenshot_20240425_193420_com google android apps photos](https://github.com/argoproj/argo-workflows/assets/148570875/dc35ca86-4120-4d07-92f1-88d57b8e7c17)

**Comment by roysha1** [comment]
@Joibel can I take this one?
